src/ng_link/dispim_link.py

```diff
@@ -84,9 +84,9 @@ def generate_dispim_link(
     tile_paths: dict[int, str] = XmlParser.extract_tile_paths(
         base_channel_xml_path
     )
-    tile_transforms: dict[
-        int, list[dict]
-    ] = XmlParser.extract_tile_transforms(base_channel_xml_path)
+    tile_transforms: dict[int, list[dict]] = XmlParser.extract_tile_transforms(
```

Comment:
instead of using list dict, please use the List and Dict coming from the typing library

Reply:
I didn't even know you could do type checking in variable assignment like this. I also changed `np.ndarray` to `np.typing.ArrayLike`